### ISSUE TITLE
Train an org wide model for Kubeflow to predict area and platform labels

### ISSUE THREAD

**Issue by jlewi** [kubeflow/code-intelligence#110]
We should train a model on all Kubeflow repositories that attempts to predict the area and platform
labels.

We already have a model for kubeflow/kubeflow but this model was last trained this summer. So
we should have significantly more data.

We'd also like to start tuning the thresholds so that more issues get area labels added automatically.

Background

* Read the [README](https://github.com/kubeflow/code-intelligence/tree/master/Label_Microservice) to understand how the issue label works

* Notebook to download GitHub issues and compute embeddings for them [https://github.com/kubeflow/code-intelligence/blob/master/Label_Microservice/notebooks/issues_loader.ipynb]
* This [notebook](https://github.com/kubeflow/code-intelligence/blob/master/Label_Microservice/notebooks/repo_mlp.ipynb) can train the model
  * The embeddings of the training data must already be computed
* This [notebook](https://github.com/kubeflow/code-intelligence/blob/master/Label_Microservice/notebooks/Training_Pipeline.ipynb) is used to launch Kubeflow pipelines to train the model

* The code to download github issues for a specific repo is [here](https://github.com/kubeflow/code-intelligence/blob/2197c647b62bcfb93e0be13d0bbfc7753fd96177/py/code_intelligence/embeddings.py#L76)

I think we can break this down into steps

1. Create a Kubelfow pipeline to download the github issues for each repo in Kubeflow
1. Add steps to the KFP pipeline to compute embeddings for the issues
1. Train a model using the combined dataset
1. Update the Kubeflow label bot to use this model to predict labels for Kubeflow

Follow on to #70

**Comment by asif001** [comment]
@jlewi I would like to work on this issue as a part of GSoC 2020. I have experience with Python, Tensorflow and Kubernetes.

**Comment by jlewi** [comment]
It looks like a good first step might be to create a notebook that does the following

1. Fetches issues for all Kubeflow repos
1. Compute embeddings for all the issues
1. Train the model on all the issues

[issues_loader.ipynb](https://github.com/kubeflow/code-intelligence/blob/master/Label_Microservice/notebooks/issues_loader.ipynb) can be used to compute the embeddings for a given repo.

[repo_mlp.ipynb](https://github.com/kubeflow/code-intelligence/blob/master/Label_Microservice/notebooks/repo_mlp.ipynb) can be used to train the model given the embeddings.

**Comment by jlewi** [comment]
#124 provides a notebook to do #1 & #2; so next step would be to train a model using those embeddings.

The embeddings I computed are stored at
gs://repo-embeddings/kubeflow/2020_0428/kubeflow_issue_embeddings_2020-04-11T17:15:10.000876-07:00.hdf5

**Comment by jlewi** [comment]
I trained the model and did some qualitative analysis by looking at predictions on recent issues. Here's an [HTML version of my notebook](https://00e9e64baca43e719d3a8eb4f403f86ca80ecaeea4c31c23eb-apidata.googleusercontent.com/download/storage/v1/b/issue-label-bot-dev_reports/o/repo_mlp.2020-0427.html?qk=AD5uMEtqXyMRuo_WEH72w74Rqy-ZYzv16xs-3lN2F4rwIlZw9SD5fAatuWu9PktoK4La53u4oza0h-oT7dfO-XGONYfxJsAFA7v3gllG6Jiwj0XiygTxVEOiOp_UnlRYrwWAvhmVS-Sl9QQusRkygJVHd35EeUeg9rhoO_TfC1SQa7O8lWHAeo9wUJ5wJaq8OkijUMWAb73Kg15ADtJ2qwtCSadU-276kjlFGk9kMlcgivQWjebH9jbp4KPWP4ciLDaD247Vg35ZwqxWFUE_2mi38fTEiWAlSYK0JNjh3unhpGszGGDzKOnYeMO3llCDl3fwhxSjMgxGZ4sxUBAuHPUPq3NXcAK46k8gM5iwydUWGZJUz-GI34V_aXFr5P3xhX_XRKdDUbdOEeGjN-1kVn502LDxXoc9pN2UhzrnDIj4G0yPaDbYR15InSbjsnSUVK_jn2PYjW0F77EQ3UuMrnquMMIG9qY8dxi1TbS_Avlq8PIMHP3yayidfINXHXm7fNu3yzQjSIc1UH0xKJEJv6bKkK6sZmXvEEIpkq3XgmjxYOvIYis1pyCg9VyDv_f5i6edet5NbRecWDC16CoOfDP1yfaklo5I8HpiPKWqw_nqWE59gUM11Ez7MDf4bLHPq-R94nxSVknmf39E2M9LpjNM37dZXwQm1HSwzf42_Fb8HSiFz6-jSJa0dBSl3uOpdaqXfQA7cgOlmsyXZ6w7kNtXoRIh3348jmx_hnmxQquYb2tDWQM0N7Vby0CjgyDSc_UyFajLHdfnqfl4nmt7IvYr3aypMq9CPSJ7Bo89J23D4Q8O0FgKEJRuwXECDINDyIOXe0_pfYcM&isca=1#Generate-Predictions-for-Recently-Created-issues)

* Only a very small fraction of issues get either a platform or area label
* Qualitatively it looks like are model is doing a poor job e.g.
   * https://github.com/kubeflow/katib/issues/1160 - "katib" appears in the issue description but we don't assign label area/katib 
   * https://github.com/kubeflow/katib/issues/1170 - "katib" appears in the title
   * https://github.com/kubeflow/kubeflow/issues/4947 - "GKE" and "GCP" appear but its not assigned label platform/gcp

**Comment by jlewi** [comment]
An org wide model is now deployed and performing very well.
